From 52c8801e4530cdcad5f97aabb722215682dad732 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Sat, 24 Aug 2019 13:31:24 +0300
Subject: [PATCH 44/52] <vm/vm_extern.h>: Move in few prototypes that are used
 in MI code.

 These are implemented by platform, but mandatory.
---
 sys/platform/vkernel64/platform/globaldata.c | 3 +--
 sys/sys/globaldata.h                         | 1 -
 sys/sys/systm.h                              | 1 -
 sys/vm/vm_extern.h                           | 4 ++++
 4 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/sys/platform/vkernel64/platform/globaldata.c b/sys/platform/vkernel64/platform/globaldata.c
index 163b2b0060..83a0e7ff72 100644
--- a/sys/platform/vkernel64/platform/globaldata.c
+++ b/sys/platform/vkernel64/platform/globaldata.c
@@ -30,8 +30,6 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
  * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
- *
- * $DragonFly: src/sys/platform/vkernel/platform/globaldata.c,v 1.5 2008/04/28 07:05:08 dillon Exp $
  */
 
 #include <sys/types.h>
@@ -41,6 +39,7 @@
 #include <sys/tls.h>
 #include <sys/proc.h>
 #include <vm/vm_page.h>
+#include <vm/vm_extern.h>
 
 #include <machine/md_var.h>
 #include <machine/globaldata.h>
diff --git a/sys/sys/globaldata.h b/sys/sys/globaldata.h
index e81faf153d..99eb765eb8 100644
--- a/sys/sys/globaldata.h
+++ b/sys/sys/globaldata.h
@@ -273,7 +273,6 @@ typedef struct globaldata *globaldata_t;
 
 #ifdef _KERNEL
 struct globaldata *globaldata_find(int cpu);
-int is_globaldata_space(vm_offset_t saddr, vm_offset_t eaddr);
 #endif
 
 #endif
diff --git a/sys/sys/systm.h b/sys/sys/systm.h
index 657e3b73f7..c4ab6b0712 100644
--- a/sys/sys/systm.h
+++ b/sys/sys/systm.h
@@ -157,7 +157,6 @@ void	cpu_vkernel_trap(struct trapframe *, int);
 enum vmm_guest_type detect_virtual(void);
 void	set_user_TLS(void);
 void	set_vkernel_fp(struct trapframe *);
-int	kvm_access_check(vm_offset_t, vm_offset_t, int);
 
 /*
  * Old CRC32 API
diff --git a/sys/vm/vm_extern.h b/sys/vm/vm_extern.h
index c6b463412f..bdbeabeb0c 100644
--- a/sys/vm/vm_extern.h
+++ b/sys/vm/vm_extern.h
@@ -135,6 +135,10 @@ kmem_alloc_stack(struct vm_map *map, vm_size_t size, int kmflags)
 	return(kmem_alloc3(map, size, VM_SUBSYS_STACK, kmflags|KM_STACK));
 }
 
+/* Implemented in platform code */
+int is_globaldata_space(vm_offset_t saddr, vm_offset_t eaddr);
+int kvm_access_check(vm_offset_t, vm_offset_t, int);
+
 #endif				/* _KERNEL */
 
 #endif				/* !_VM_VM_EXTERN_H_ */
-- 
2.22.0

